modules/javafx.controls/src/test/java/test/javafx/scene/control/ListCellTest.java

```diff
@@ -1130,6 +1135,56 @@ public void testEditCommitValueChangeIsReflectedInCell() {
         assertEquals("ABCDEF [Changed]", cell.getItem());
     }
 
+    /**
+     * Same index and underlying item should not cause the updateItem(..) method to be called.
+     */
+    @Test
+    public void testSameIndexAndItemShouldNotUpdateItem() {
+        AtomicInteger counter = new AtomicInteger();
+
+        list.setCellFactory(e -> new ListCell<>() {
+            @Override
+            protected void updateItem(String item, boolean empty) {
+                super.updateItem(item, empty);
+
+                counter.incrementAndGet();
```

Comment:
very, very minor, and probably not important: shouldn't 'incrementAndGet()` be called before `super.updateItem()`?

Reply:
Hm, I see no reason for either one or the other. So not strong preference from my side here.